Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -1860,4 +1862,128 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::to_sorted)
+        actual_delete_count = 0;
+    }
+    // 9. Else if deleteCount is not present, then
+    else if (vm.argument_count() == 1) {
+        // a. Let actualDeleteCount be len - actualStart.
+        actual_delete_count = length - actual_start;
+    }
+    // 10. Else,
+    else {
+        // a. Let dc be ? ToIntegerOrInfinity(deleteCount).
+        auto dc = TRY(delete_count.to_integer_or_infinity(global_object));
+
+        // b. Let actualDeleteCount be the result of clamping dc between 0 and len - actualStart.
+        actual_delete_count = static_cast<size_t>(clamp(dc, 0, static_cast<double>(length - actual_start)));
+    }
+
+    // Sanity check
+    VERIFY(actual_delete_count <= (length - actual_start));
+
+    // 11. Let newLen be len + insertCount - actualDeleteCount.
+    auto new_length_double = static_cast<double>(length) + static_cast<double>(insert_count) - static_cast<double>(actual_delete_count);
+
+    // 12. If newLen > 2^53 - 1, throw a TypeError exception.
+    if (new_length_double > MAX_ARRAY_LIKE_INDEX)
+        return vm.throw_completion<TypeError>(global_object, ErrorType::ArrayMaxSize);
+
+    auto new_length = static_cast<size_t>(new_length_double);
+
+    // 13. Let A be ? ArrayCreate(𝔽(newLen)).
+    auto* array = TRY(Array::create(global_object, new_length));
+
+    // 14. Let i be 0.
+    size_t i = 0;
+
+    // 15. Let r be actualStart + actualDeleteCount.
+    auto r = actual_start + actual_delete_count;
+
+    // 16. Repeat, while i < actualStart,
+    for (; i < actual_start; ++i) {
+        // a. Let Pi be ! ToString(𝔽(i)).
+        auto property_key = PropertyKey { i };
+
+        // b. Let iValue be ? Get(O, Pi).
+        auto i_value = TRY(object->get(property_key));
+
+        // c. Perform ! CreateDataPropertyOrThrow(A, Pi, iValue).
+        MUST(array->create_data_property_or_throw(property_key, i_value));
+
+        // d. Set i to i + 1.
+    }
+
+    // 17. For each element E of items, do
+    for (size_t element_index = 2; element_index < vm.argument_count(); ++element_index, ++i) {
+        auto element = vm.argument(element_index);
+
+        // a. Let Pi be ! ToString(𝔽(i)).
+        auto property_key = PropertyKey { i };
+
+        // b. Perform ! CreateDataPropertyOrThrow(A, Pi, E).
+        MUST(array->create_data_property_or_throw(property_key, element));
+
+        // c. Set i to i + 1.
+    }
+
+    // 18. Repeat, while i < newLen,
+    for (; i < new_length; ++i, ++r) {
+        // a. Let Pi be ! ToString(𝔽(i)).
+        auto property_key = PropertyKey { i };
+
+        // b. Let from be ! ToString(𝔽(r)).
+        auto from = PropertyKey { r };
+
+        // c. Let fromValue be ? Get(O, from).
+        auto from_value = TRY(object->get(from));
+
+        // d. Perform ! CreateDataPropertyOrThrow(A, Pi, fromValue).
+        MUST(array->create_data_property_or_throw(property_key, from_value));
+
+        // e. Set i to i + 1.
+        // f. Set r to r + 1.
```

Comment:
Similar comment here with `++r`, definitely not needed maybe it is more clear here because we also do i to i + 1